Source/Core/Duality/Utility/Coroutines/CoroutineManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// The Coroutine Manager
+	/// </summary>
+	public class CoroutineManager
+	{
+		private readonly Queue<Coroutine> pool = new Queue<Coroutine>(64);
+		private readonly Queue<Coroutine> coroutines = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> nextCycle = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> scheduled = new Queue<Coroutine>();
+		private readonly Queue<Coroutine> trashcan = new Queue<Coroutine>();
+
+		private readonly Dictionary<Coroutine, Exception> lastFrameErrors = new Dictionary<Coroutine, Exception>();
+
+		private int count;
+
+		/// <summary>
+		/// Returns an IEnumerable of all currently active and scheduled Coroutines
+		/// </summary>
+		public IEnumerable<Coroutine> Coroutines
+		{
+			get { return this.coroutines.Concat(this.scheduled); }
+		}
+
+		/// <summary>
+		/// Returns an IEnumerable of the Exception encountered in the last cycle, with the relative Coroutine
+		/// </summary>
+		public IEnumerable<KeyValuePair<Coroutine, Exception>> LastFrameErrors
```

Comment:
We should expose this as a ReadonlyDictionary so it not possible to modify it anymore

Reply:
Will do